client/ayon_maya/plugins/publish/extract_layout.py

```diff
@@ -84,80 +85,81 @@ def process(self, instance):
-            container = containers[0]
 
-            representation_id = cmds.getAttr(
-                "{}.representation".format(container))
-
-            # Ignore invalid UUID is the representation for whatever reason
-            # is invalid
-            if not is_valid_uuid(representation_id):
-                self.log.warning(
-                    f"Skipping container with invalid UUID: {container}")
-                continue
-
-            # TODO: Once we support managed products from another project
-            #  we should be querying here using the project name from the
-            #  container instead.
-            representation = get_representation_by_id(
-                project_name,
-                representation_id,
-                fields={"versionId", "context", "name"}
-            )
-            if not representation:
-                self.log.warning(
-                    "Representation not found in current project "
-                    "for container: {}".format(container))
-                continue
-
-            version_id = representation["versionId"]
-            # TODO use product entity to get product type rather than
-            #    data in representation 'context'
-            repre_context = representation["context"]
-            product_type = repre_context.get("product", {}).get("type")
-            if not product_type:
-                product_type = repre_context.get("family")
-
-            json_element = {
-                "product_type": product_type,
-                "instance_name": cmds.getAttr(
-                    "{}.namespace".format(container)),
-                "representation": str(representation_id),
-                "version": str(version_id),
-                "extension": repre_context["ext"],
-                "host": self.hosts
-            }
-
-            local_matrix = cmds.xform(asset, query=True, matrix=True)
-            local_rotation = cmds.xform(asset, query=True, rotation=True, euler=True)
-
-            t_matrix = self.create_transformation_matrix(local_matrix, local_rotation)
-
-            json_element["transform_matrix"] = [
-                list(row)
-                for row in t_matrix
-            ]
-
-            basis_list = [
-                1, 0, 0, 0,
-                0, 1, 0, 0,
-                0, 0, 1, 0,
-                0, 0, 0, 1
-            ]
-
-            basis_mm = om.MMatrix(basis_list)
-            b_matrix = convert_matrix_to_4x4_list(basis_mm)
-
-            json_element["basis"] = []
-            for row in b_matrix:
-                json_element["basis"].append(list(row))
-
-            json_element["rotation"] = {
-                "x": local_rotation[0],
-                "y": local_rotation[1],
-                "z": local_rotation[2]
-            }
-            json_data.append(json_element)
+            container_dict = self.get_containers_sub_assembly(containers, asset)
+            for container, asset in container_dict.items():
```

Comment:
The `asset` variable here shadows the `asset` variable in the outer scope above, overriding it. Please provide a new unique variable name.

Also, everytime I read this code I'm very confused about what this _asset_ is - but all I can see is that it's intended to represent the "root" node transform of the container? I wonder if we should improve the variable somewhere or add some comments about this in the code.

Reply:
`asset` variable mean the loaded asset which is hooked to the container data. maybe we can rename as `loaded_asset`

Reply:
Still unsure what that means? What is the "loaded asset" - I have a feeling you're describing something in the scene but I have no knowledge myself of anything in Maya being known as 'an asset'. To me `asset` sounds very much like OpenPype "folder" and maybe it's actually referring to the folder? (which I don't think it does looking at the code).

To me it's more of a `container_root`, or `container_transform`. Also, note that _technically_ something loaded in _may_ have multiple root nodes, but I don't believe we're supporting that in the Layout Publishing and are always assuming loaded grouped containers where we use the 'group'  as the root node for that container.